sdk/nodejs/docker.ts

```diff
@@ -292,9 +284,6 @@ async function pullCacheAsync(
 
     pulumi.log.debug(`pulling cache for ${imageName} from ${repoUrl}`, logResource);
 
-    // Ensure that we're logged in to the source registry and attempt to pull each stage in turn.
-    await login();
-
```

Comment:
we would only call login here if we both wanted to pull from the cache and if we had a repoUrl.  Those two checks are now done above.